bundles/org.openhab.binding.nanoleaf/src/main/java/org/openhab/binding/nanoleaf/internal/OpenAPIUtils.java

```diff
@@ -66,28 +78,42 @@ public static Request requestBuilder(HttpClient httpClient, NanoleafControllerCo
         }
         URI requestURI;
         try {
-            requestURI = new URI(HttpScheme.HTTP.asString(), null, address, port, path, null, null);
+            requestURI = new URI(HttpScheme.HTTP.asString(), null, address, port, path, query, null);
         } catch (URISyntaxException use) {
-            logger.warn("URI could not be parsed with path {}", path);
+            LOGGER.warn("URI could not be parsed with path {}", path);
             throw new NanoleafException("Wrong URI format for API request");
         }
-        return httpClient.newRequest(requestURI).method(method);
+        return requestURI;
     }
 
     public static ContentResponse sendOpenAPIRequest(Request request)
             throws NanoleafException, NanoleafUnauthorizedException {
         try {
+            LOGGER.trace("Sending Request {} {}",request.getURI(), request.getQuery() == null ? "no query parameters": request.getQuery());
```

Comment:
It seems this whole block is about trace logging, I think the best thing would be to extract a method passing the `request` and returning immediately if trace logging is not enabled and otherwise execute the statements.

Reply:
extracted as method and only traces if trace is enabled.